src/psyclone/tests/psyir/transformations/increase_rank_loop_arrays_test.py

```diff
@@ -80,25 +79,63 @@ def test_irla_validate(fortran_reader):
-        trans.apply(routine.children[1], options={'arrays': ['i']})
-    assert ("IncreaseRankLoopArraysTrans provided 'arrays' must be a local "
+        trans.apply(routine.children[1], arrays=['i'])
+    assert ("IncreaseRankLoopArraysTrans provided 'arrays' must be local "
             "array symbols, but 'i: " in str(err.value))
 
     with pytest.raises(TransformationError) as err:
-        trans.apply(routine.children[1], options={'arrays': ['a']})
+        trans.apply(routine.children[1], arrays=['a'])
     assert ("IncreaseRankLoopArraysTrans does not support arrays that are "
             "referenced inside a Codeblock, but 'a' is inside one."
             in str(err.value))
 
-    # Referencing non-existan arrays is fine, they are just ignored, this
-    # is useful to apply the transformation to several routines without an
-    # exact match of symbols
-    trans.apply(routine.children[1], options={'arrays': ['non_existant']})
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.children[1], arrays=['non_existant'])
+    assert ("IncreaseRankLoopArraysTrans provided array 'non_existant' does "
+            "not existin this scope." in str(err.value))
 
     with pytest.raises(TransformationError) as err:
-        trans.apply(routine.children[1].detach(), options={'arrays': ['a']})
-    assert ("The target of the IncreaseRankLoopArraysTrans transformation "
-            "should be a Loop inside a Routine." in str(err.value))
+        trans.apply(routine.children[1].detach(), arrays=['a'])
+    assert ("The target Loop of the IncreaseRankLoopArraysTrans transformation"
+            " must be inside a Routine." in str(err.value))
+
+
+def test_irla_validate_bounds(fortran_reader):
+    ''' Check that the validate method checks if the loop bounds are static
+    when possible. '''
+    trans = IncreaseRankLoopArraysTrans()
+    psyir = fortran_reader.psyir_from_source("""
+        program test
+            use other
+            real, dimension(10) :: a
+            integer, parameter :: constant = 3
+            integer :: i, j
+            variable = 3
+
+            ! This loop has variable bounds
+            do j = 1, variable
+                do i = 1, 10
+                   a(i) = 3
+                end do
+            enddo
+
+            ! These could be static
+            do j = size(variable), constant + imported
+                do i = 1, 10
+                   a(i) = 3
+                end do
+            enddo
+        end program
+    """)
+    routine = psyir.children[0]
+    with pytest.raises(TransformationError) as err:
+        trans.apply(routine.walk(Loop)[0], arrays=['a'])
+    assert ("IncreaseRankLoopArraysTrans can only be applied to loops with "
+            "static loop bound expressions, but in has been attempted "
+            "in a loop with the variable 'variable'." in str(err.value))
+
+    # This is fine
+    trans.apply(routine.walk(Loop)[2], arrays=['a'])
```

Comment:
Actually, I suspect it isn't - please check the output.

Reply:
Oh did you mean because the ArrayBound missed the dangling_parent to avoid the "linebreak" in the declaration?

Reply:
> So the first loop now writes to a(i,:) every trip which isn't great.

@arporter The example may be a bit unfortunate but the original first loop already writes to `a(i)` multiple time:
```
            do j = 1, variable
                do i = 1, 10
                   a(i) = 3
                end do
            enddo
```
so the transformed code just does the same but for the whole slice `a(i,:)`